src/main/kotlin/ch/unibas/dmi/dbis/cottontail/model/values/ComplexVectorValue.kt

```diff
@@ -0,0 +1,113 @@
+package ch.unibas.dmi.dbis.cottontail.model.values
+
+import ch.unibas.dmi.dbis.cottontail.model.values.complex.Complex
+import ch.unibas.dmi.dbis.cottontail.model.values.complex.ComplexArray
+
+/**
+ * This is an abstraction over a [ComplexArray] and it represents a vector of [Complex]s.
+ *
+ * @author Manuel Huerbin
+ * @version 1.0
+ */
+inline class ComplexVectorValue(override val value: ComplexArray) : VectorValue<ComplexArray> {
```

Comment:
Same as for the **ComplexValue** implementation. Treat a the **ComplexVectorValue** implementation as an array of floats/doubles. If the _complex array_ has x components, you need 2 * x components in the underlying array.